### PR TITLE
TTPassGen

### PR DESCRIPTION
**Pull request by dualfade**
New tool request --
fix 2757

### CODE REVIEW

File: lists/to-release [hunk 1]
```diff
@@ -1 +1 @@
-pspy
+ttpassgen
```
Comment (noraj) [inline lists/to-release:1]: Don't replace, just add to the list.

File: packages/ttpassgen/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+pkgname=TTPassGen
```
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:4]: always downcase in pkgname. you can add a `_pkgname` var if you need to use the mixed case name somewhere in the script. So do:

```
pkgname=ttpassgen
_pkgname=TTPassGen
```

File: packages/ttpassgen/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+pkgver=133.a06d99d
+pkgrel=1
+pkgdesc='is a highly flexiable and scriptable password dictionary generator base on Python, you can easily use various rules to generate the desired combination of words.'
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:7]: pkgdesc='Highly flexiable and scriptable password dictionary generator base on Python, you can easily use various rules to generate the desired combination of words.'

File: packages/ttpassgen/PKGBUILD (hunk 2 above)
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:4]: `pkgname` should be lowercase.

File: packages/ttpassgen/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+arch=('any')
+groups=('blackarch' 'blackarch-automation' 'blackarch-cracker')
+url='https://github.com/tp7309/TTPassGen'
+license=('Apache')
+depends=('python' 'python-tqdm' 'python-click')
+makedepends=('git' 'python-setuptools')
+source=("git+https://github.com/tp7309/TTPassGen.git")
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:14]: `source=("git+https://github.com/tp7309/$pkgname.git")`

File: packages/ttpassgen/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+sha512sums=('SKIP')
+
+build() {
+    cd "$pkgname"
+    python setup.py build
+}
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:20]:
```
build() {
    cd $pkgname

    python setup.py build
}
```

File: packages/ttpassgen/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+package() {
+    cd "$pkgname"
+    python setup.py install --root="$pkgdir" --prefix=/usr -O1 --skip-build
+
+    install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
+    install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+}
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:28]:
```
package() {
    cd "$pkgname"

    python setup.py install --root="$pkgdir" --prefix=/usr -O1 --skip-build
    install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
    install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
}
```

File: packages/ttpassgen/PKGBUILD (hunk 3 above)
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:7]: Begins with a capitalized sentence and omitting the tool name or saying `is a`: eg: `Highly flexible and scriptable password dictionary generator.`.

Don't put the language, keep it short and in general try not to re-use the tool disclaimer unless it's well written.

By the way the upstream desc has a typo:

```diff
-flexiable
+flexible
```

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]

File: lists/to-release (hunk 1 above)
Comment (Edu4rdSHL) [inline lists/to-release:1]: Do not delete pspy, please.

File: packages/ttpassgen/PKGBUILD (hunk 4 above)
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:14]: Use `$_pkgname` here. No var in url but var in source is ok.

File: packages/ttpassgen/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+groups=('blackarch' 'blackarch-automation' 'blackarch-cracker')
+url='https://github.com/tp7309/TTPassGen'
+license=('Apache')
+depends=('python' 'python-tqdm' 'python-click')
+makedepends=('git' 'python-setuptools')
+source=("git+https://github.com/tp7309/TTPassGen.git")
+sha512sums=('SKIP')
+
+build() {
+    cd "$pkgname"
```
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:18]: You'll need to fix into `cd $_pkgname`.

File: packages/ttpassgen/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    python setup.py build
+}
+
+package() {
+    cd "$pkgname"
```
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:23]: You'll need to fix into `cd $_pkgname`.

File: packages/ttpassgen/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    python setup.py install --root="$pkgdir" --prefix=/usr -O1 --skip-build
+
+    install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
+    install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
```
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:27]: Apache license is already in `/usr/share/licenses/common/Apache` so don't include it. Follow https://wiki.archlinux.org/index.php/PKGBUILD#license for licenses.

File: packages/ttpassgen/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+
+
```
Comment (noraj) [inline packages/ttpassgen/PKGBUILD:31]: Keep only one blank line at the end.

**Review by noraj — CHANGES_REQUESTED** [review]

File: lists/to-release (hunk 1 above)
Comment (dualfade) [reply lists/to-release:1]: got it. re added. Thank you for that note.

File: packages/ttpassgen/PKGBUILD (hunk 2 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:4]: Will include $_pkgname as url is capitalized. Thank you for this note. Maybe update template ?

File: packages/ttpassgen/PKGBUILD (hunk 3 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:7]: Corrected spelling.

File: packages/ttpassgen/PKGBUILD (hunk 4 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:14]: Got it. Thank you for clarification.

File: packages/ttpassgen/PKGBUILD (hunk 7 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:18]: Check.

File: packages/ttpassgen/PKGBUILD (hunk 8 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:23]: Check.

File: packages/ttpassgen/PKGBUILD (hunk 9 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:27]: Again. Thank you for clarification.

File: packages/ttpassgen/PKGBUILD (hunk 10 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD:31]: Got it.Again thank you for clarification on this as well.
Will push update.

File: packages/ttpassgen/PKGBUILD [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+groups=('blackarch' 'blackarch-automation' 'blackarch-cracker')
+url='https://github.com/tp7309/TTPassGen'
+license=('Apache')
+depends=('python' 'python-tqdm' 'python-click')
+makedepends=('git' 'python-setuptools')
+source=("git+https://github.com/tp7309/$_pkgname")
+sha512sums=('SKIP')
+
+build() {
+    cd "$_pkgname"
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:19]: Remove quotes.

File: packages/ttpassgen/PKGBUILD [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    python setup.py build
+}
+
+package() {
+    cd "$_pkgname"
```
Comment (Edu4rdSHL) [inline packages/ttpassgen/PKGBUILD:24]: Remove quotes.

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]

File: packages/ttpassgen/PKGBUILD (hunk 11 above)
Comment (dualfade) [reply packages/ttpassgen/PKGBUILD]: adjusting. Maybe update template on this as well ? Thanks.

Comment (Edu4rdSHL) [reply packages/ttpassgen/PKGBUILD]: For some reasons, quotes are a good practice for some tools that have special characters or spaces in their name but if there's none of the mentioned above then quotes aren't needed.

**Comment by Edu4rdSHL** [comment]
Thank you, @dualfade

File: packages/ttpassgen/PKGBUILD (hunk 2 above)
Comment (noraj) [reply packages/ttpassgen/PKGBUILD]: In most case pkgname = _pkgname so no.

File: packages/ttpassgen/PKGBUILD (hunk 11 above)
Comment (noraj) [reply packages/ttpassgen/PKGBUILD]: For example quote are needed when `cd $_pkgname-$pkgver"` because of the dash.

**Pull request merged**